### ISSUE TITLE
postcss.config.js modifications

### ISSUE THREAD

**Issue by rossberenson** [10up/10up-toolkit#165]
### Describe your question

Trying to create mixins via js. I created a `postcss.config.js`. I was able to add plugins to it, which works great. I attached the code I'm using. I added a very simple mixin for debugging purposes. I know this file is being read since the globbing and nesting does work. However, when I try to add mixins to postcss-mixins which has been declared in the toolkit, it doesn't work. Perhaps I'm not modifying it correctly? I've tried a few different ways but no luck. 

Customizing method from https://github.com/10up/10up-toolkit/blob/develop/packages/toolkit/README.md#customizing-postcss.

When using `@mixin bgcolor;` the output in dist/style.css is:

```
.class {
    @mixin bgcolor;
}
```
When I think it should be:
```
.class {
    background-color: 'red';
}
```

postcss.config.js
```
const baseConfig = require('10up-toolkit/config/postcss.config');
require('postcss-import-ext-glob');
require('postcss-nested');

module.exports = (props) => {
	const config = baseConfig(props);

	config.plugins['postcss-import-ext-glob'] = {
		sort: 'asc',
	};

	config.plugins['postcss-nested'] = {
		preserveEmpty: false,
	};

	config.plugins['postcss-mixins'] = {
		mixins: {
			bgcolor(mixin, color = 'red') {
			 	return {
			 		'background-color': color,
			 	};
			},
		},
	};

	return config;
};
```

Any assistance would be greatly appriciated. I've been researching to see if there's a bug somewhere but I haven't come across anything. Looking through the toolkit postcss.config.js, it looks right. It's probably 99.9% me. haha

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nicholasio** [comment]
10up-toolkit uses postcss-preset-env by default and therefore it comes with `postcss-nesting` instead of `postcss-nested`. Do you need `postcss-nested`? I don't think you can have both, maybe that's what is breaking it.

If you do need `postcss-nested` you might need to remove `postcss-preset-env`.

@Antonio-Laguna Any thoughts on this?

**Comment by rossberenson** [comment]
I don't need both. I removed `postcss-nested`, the mixin challenge still exists. Not like this was the solution, just stating that there was no conflict that is causing the mixin not to work.

**Comment by Antonio-Laguna** [comment]
Hey @rossberenson I've tested this like this:

```js
const baseConfig = require('10up-toolkit/config/postcss.config.js');

module.exports = (props) => {
	const config = baseConfig(props);

	config.plugins['postcss-mixins'] = {
		mixins: {
			bgcolor(mixin, color = 'red') {
				return {
					'background-color': color,
				};
			},
		},
	};

	return config;
};
```

And with this CSS

```pcss
.test {
  @mixin bgcolor;
}
```

I do get:

```css
.test{
  background-color: red;
}
```

This means there's something I'm missing 🤔 I've even tried using the mixin within an imported file and I see the output correctly too.

If I have to suspect, there's something with the order of the plugins and `postcss-import-ext-glob`. Could you try with this file? 

```js
const baseConfig = require('10up-toolkit/config/postcss.config');

module.exports = (props) => {
	const config = baseConfig(props);

	config.plugins = {
		'postcss-import-ext-glob': {
			sor: 'asc',
		},
		...config.plugins,
	};

	config.plugins['postcss-mixins'] = {
		mixins: {
			bgcolor(mixin, color = 'red') {
				return {
					'background-color': color,
				};
			},
		},
	};

	return config;
};
```

What we're doing is to make sure that the object has `postcss-import-ext-glob` first as their [usage](https://github.com/dimitrinicolas/postcss-import-ext-glob#usage) implies.

You don't need to add the to `require` but rather just make sure that those dependencies are installed.

If this doesn't work I'd need to check with your current code, could you point me to a repo or prepare something minimal for me to check?

Keep us posted!

---

Side note,  you can keep `postcss-nested` if you prefer Sass-like nesting but I'd disable that on `postcss-preset-env` like this: 

```js
config.plugins['postcss-preset-env'].features['nesting-rules'] = false;
```

**Comment by rossberenson** [comment]
@Antonio-Laguna - That was the solution. Funny, as I was winding down for bed last night, like everyone else, my mind was running a million miles per hour and I realized it could be the globbing. That's the first thing I wanted to debug when I woke up. Spreading `...config.plugins,` after the glob did the trick.  Funny because a while back, I did try something similar but now realizing I was spreading within the plugin's config and not the plugin array.

I greatly appreciate both yours and @nicholasio's help with this. Always learning. Thanks for the very thoughtful and informative response.
